roles/junit2json/tasks/merge.yml

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: Apache-2.0
+---
+# tasks file merge.yml for role redhatci.ocp.junit2json
+# merges multiple XMLs into 1 and converts it into JSON
+
+- name: Generate XML files list
+  ansible.builtin.set_fact:
+    junit2json_reports_list: "{{ junit2json_reports_list | default([]) + [item] }}"
+  loop: "{{ _junit2json_found_reports.files | map(attribute='path') }}"
+
+# debug only:
+- name: Print list of XML reports file names
+  ansible.builtin.debug:
+    msg:
+      - "XML files names: {{ junit2json_reports_list }}"
+  when:
+    - junit2json_debug is defined
+    - junit2json_debug
+
+- name: Ensure junitparser is installed
+  ansible.builtin.command:
+    cmd: |
+      command -v junitparser
+  register: _junit2json_junitparser_command
+  changed_when:
+    - true
+
+- name: Validate some variables
```

Comment:
Hi Max, 
Few comments here please - 

1. The task name does not specify the real intention of this task. 
2. I would put this assertion somewhere at the beginning of the playbook, so that if this package is missing it will fail on an early stage (as @thekad) wrote. 
3. To me it looks redundant to register a variable to represent package installation, there are more efficient, readable methods to do this, WDYT?